7-streaming-endpoint/main.py

```diff
@@ -0,0 +1,57 @@
+from pydantic import BaseModel
+from typing import Optional
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    GenerationConfig,
+    TextIteratorStreamer,
+)
+import torch
+
+modal_path = "tiiuae/falcon-7b-instruct"
+
+# Loading in base model and tokenizer
+tokenizer = AutoTokenizer.from_pretrained(modal_path)
+tokenizer.pad_token = tokenizer.eos_token
+model = AutoModelForCausalLM.from_pretrained(
+    modal_path,
+    torch_dtype=torch.bfloat16,
+    trust_remote_code=True,
+    device_map="auto",
+)
+model.to("cuda")
+
+class Item(BaseModel):
+    prompt: str
+    cutoff_len: Optional[int] = 256
+    temperature: Optional[float] = 0.8
+    top_p: Optional[float] = 0.75
+    top_k: Optional[float] = 40
+    max_new_tokens: Optional[int] = 250
+
+def predict(item, run_id, logger):
+    item = Item(**item)
+    inputs = tokenizer(
+      item.prompt, return_tensors="pt", max_length=512, truncation=True, padding=True
+    )
+    input_ids = inputs["input_ids"].to("cuda")
+
+    streamer = TextIteratorStreamer(tokenizer)
```

Comment:
So the user needs a new streamer obj for each predict run? 
Might be a potential memory leak if it’s not cleaned up nicely by transformers